waku/factory/app.nim

```diff
@@ -456,18 +314,15 @@ proc setupMonitoringAndExternalInterfaces*(app: var App): AppResult[void] =
       app.metricsServer = some(startMetricsServerRes.value)
 
   if app.conf.metricsLogging:
-    let startMetricsLoggingRes = startMetricsLogging()
-    if startMetricsLoggingRes.isErr():
-      error "Starting metrics console logging failed. Continuing in current state.",
-        error = startMetricsLoggingRes.error
+    startMetricsLog()
 
   ok()
 
 # App shutdown
 
 proc stop*(app: App): Future[void] {.async: (raises: [Exception]).} =
-  if app.restServer.isSome():
-    await app.restServer.get().stop()
+  if app.conf.rest:
+    await app.restServer.stop()
```

Comment:
Not sure I don't understand this change. Shouldn't we check for the `restServer` to be already mounted and not just in the configurations?

Reply:
Yes makes sense! I'll double-check